### PR TITLE
Fix caching and disable by default

### PR DESCRIPTION
**Pull request by ingwinlu**
this is a bugfixing pr resulting from my work in #1745  and just fixes existing errors and bugs

adds changes discussed in #1689

### CODE REVIEW

**Comment by justinmayer** [comment]
Many thanks, Winlu. Much appreciated.

Any comments on this from other folks? (cc: @kylef / @smartass101 / @avaris / @iKevinY)

File: pelican/__init__.py [hunk 1]
```diff
@@ -101,6 +101,12 @@ def _handle_deprecation(self):
                             'PAGE_LANG_URL'):
                 logger.warning("%s = '%s'", setting, self.settings[setting])
 
+        if self.settings.get('AUTORELOAD_IGNORE_CACHE'):
+            logger.warning('Found deprecated `AUTORELOAD_IGNORE_CACHE` in '
+                           'settings. It has no effect now and can be safely '
+                           'removed.')
```
Comment (avaris) [inline pelican/__init__.py]: Umm, why? People can still enable cache and might want to disable it in ``--autoreload``.

**Comment by iKevinY** [comment]
Docstrings should probably use `"""` rather than `'''` for the sake of consistency.

**Comment by ingwinlu** [comment]
@iKevinY 
I did not touch those, they are the same as they were in utils.py (there we use ')

I am hesitant to do a unification in this pr as it is not strictly part of it, but I agree with you that we should do a pep8 pr soon (maybe even before the next release)

**Comment by iKevinY** [comment]
@ingwinlu Yeah, I realized that after reading through the changes more closely. I think it would be safe to make that change in this PR since it's purely syntactic.

Definitely agree on the need for a style-standardization pull request in the near future.

**Comment by ingwinlu** [comment]
@avaris 
The same behaviour without that setting can still be archived via `--autoreload --ignore-cache`. So no loss of functionality.

personal opinion:
settings.py is pretty packed, people might look at it and not find what they are looking for. Cache alone had 7. I don't feel like the level of customization offered is needed at all. Any setting we can remove is a good thing (less complexity, less to document, less to break)

**Comment by ingwinlu** [comment]
@iKevinY flake8 gently reminded me that i forgot a hashlib import, so definitely was worth checking style ;)

**Comment by avaris** [comment]
@ingwinlu, I'm fine with that. But, I think it would be better if we mention ``--ignore-cache`` in the deprecation warning. Something like:

```python
logger.warning('Found deprecated `AUTORELOAD_IGNORE_CACHE` in ' 
               'settings. Use --ignore-cache option instead.')
```

**Comment by ingwinlu** [comment]
When everyone is happy with the review commits I will squash them into the 2 main ones and we should be good to go

**Comment by avaris** [comment]
:+1:

**Comment by justinmayer** [comment]
Excellent work, Winlu.  (^_^)

**Pull request merged**